js/game.js

```diff
@@ -9,8 +11,80 @@ function whack(rat) {
   rat.classList.add('hidden')
 }
 
+/**
+ * Hide all the rats.
+ *
+ * @param {Nodelist} rats All the rats.
+ */
+function hideRats(rats) {
+    rats.forEach(function(rat) {
+        rat.classList.add('hidden')
+    })
+}
+
 rats.forEach(function(rat) {
   rat.addEventListener('click', function() {
     whack(rat)
   })
 })
+
+/**
+ * this function randomises between the times entered
+ * @param times (amount of time)
+ * @returns one of the times entered at random
+ */
+function random_time(times) {
+  return times[Math.floor(Math.random()*times.length)];
+}
+
+/**
+ * function selects random mole displays it and hides it after a defined period of time
+ */
+function showRat(){
+    var rats = document.querySelectorAll('.rat.hidden')
+    if (rats.length > 0 ) {
+        var randomRat = rats[Math.floor(Math.random() * rats.length)]
```

Comment:
this line uses the same logic as the random_time function. if you rename that function to something more generic you could use it to pick a random rat